CDCgov/prime-reportstream
# Bump @rest-hooks/rest from 3.0.3 to 5.0.1 in /frontend-react

This PR we upgrade @rest-hooks/rest to 5.0.1 (like https://github.com/CDCgov/prime-reportstream/pull/5108 but with breaking changes addressed)

[useFetchInit()](https://resthooks.io/docs/api/HookableResource#useFetchInit) -> [getFetchInit()](https://resthooks.io/docs/api/resource#getFetchInit)

`useFetchInit()` should only be used if hooks are needed, and in @rest-hooks/rest this was moved out to [HookableResource](https://resthooks.io/docs/api/HookableResource) to better enforce hook usage.

[getFetchInit()](https://resthooks.io/docs/api/resource#getFetchInit) does the same thing but is called when the fetch actually happens.

Test Steps:
1. Visit any pages that do auth based request
2. Ensure auth context is included in fetch

## Changes
- Upgrade @rest-hooks/rest to 5.0.1
- Move AuthResource's useFetchInit to getFetchInit to ensure compatibility.

## Checklist

### Testing

### Process

## Fixes
- #issue - List GitHub issues this PR fixes

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
